stock_helper_delivery/models/stock_picking.py

```diff
@@ -17,15 +24,21 @@ def get_picking_with_carrier_from_chain(self):
         if self.carrier_id:
             return self
 
-        def get_first_move_dest(moves):
-            for move in moves.move_dest_ids:
-                if move.state not in ("cancel", "done"):
-                    return move
-
         move_dest = get_first_move_dest(self.move_lines)
         while move_dest:
             if move_dest.picking_id.carrier_id:
                 return move_dest.picking_id
             move_dest = get_first_move_dest(move_dest)
         # Should return an empty record if we reach this line
         return self.browse()
+
+    def _get_ship_from_chain(self, done=False):
+        """Returns the shipment related to the current operation."""
+        move_dest = get_first_move_dest(self.move_lines, done=done)
+        while move_dest:
+            picking = move_dest.picking_id
+            if picking.picking_type_id.code == "outgoing":
+                return picking
+            move_dest = get_first_move_dest(move_dest, done=done)
+        # Should return an empty record if we reach this line
+        return self.browse()
```

Comment:
Should it not return an empty move instead of stock.picking ?

Reply:
No this method returns a `stock.picking` record (the delivery one if we found one), it never returns a move.

Reply:
:+1: